core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/recursively.kt

```diff
@@ -0,0 +1,305 @@
+     *
+     * `df.`[select][DataFrame.select]` { myColumnGroup.`[all][ColumnsSelectionDsl.all]`().`[rec][TransformableColumnSet.rec]`() }`
+     *
+     * `df.`[select][DataFrame.select]` { `[groups][ColumnsSelectionDsl.groups]`().`[recursively][TransformableColumnSet.recursively]`() }`
+     */
+    public fun <C> TransformableColumnSet<C>.recursively(): ColumnSet<C> =
+        recursivelyImpl(includeTopLevel = true, includeGroups = true)
+
+    /** @include [TransformableColumnSet.recursively] */
+    public fun <C> TransformableColumnSet<C>.rec(): ColumnSet<C> = recursively()
+
+    /**
+     * @include [CommonRecursivelyDocs]
+     * @arg [CommonRecursivelyDocs.Examples]
+     *
+     * `df.`[select][DataFrame.select]` { `[first][ColumnsSelectionDsl.firstChild]` { col -> col.`[any][DataColumn.any]` { it == "Alice" } }.`[recursively][TransformableSingleColumn.recursively]`() }`
+     *
+     * `df.`[select][DataFrame.select]` { `[single][ColumnsSelectionDsl.singleChild]` { it.name == "myCol" }.`[rec][TransformableSingleColumn.rec]`() }`
+     */
+    public fun TransformableSingleColumn<*>.recursively(): SingleColumn<*> =
+        recursivelyImpl(includeTopLevel = true, includeGroups = true)
+
+    /** @include [TransformableSingleColumn.recursively] */
+    public fun TransformableSingleColumn<*>.rec(): SingleColumn<*> = recursively()
+
+    // endregion
+
+    // region dfs
+
+    @Deprecated(
+        message = "dfs is deprecated, use recursively instead.",
+        replaceWith = ReplaceWith("this.cols(predicate).recursively(includeTopLevel = false)"),
+        level = DeprecationLevel.WARNING,
+    )
+    public fun <C> ColumnSet<C>.dfs(predicate: (ColumnWithPath<*>) -> Boolean): ColumnSet<*> = dfsInternal(predicate)
+
+    @Deprecated(
+        message = "dfs is deprecated, use recursively instead.",
+        replaceWith = ReplaceWith("this.cols(predicate).recursively()"),
+        level = DeprecationLevel.WARNING,
+    )
+    public fun SingleColumn<DataRow<*>>.dfs(predicate: (ColumnWithPath<*>) -> Boolean): ColumnSet<*> =
+        ensureIsColGroup().asColumnSet().dfsInternal(predicate)
+
+    @Deprecated(
+        message = "dfs is deprecated, use recursively instead.",
+        replaceWith = ReplaceWith("this.cols(predicate).recursively()"),
+        level = DeprecationLevel.WARNING,
+    )
+    public fun ColumnsSelectionDsl<*>.dfs(predicate: (ColumnWithPath<*>) -> Boolean): ColumnSet<*> =
+        asSingleColumn().dfs(predicate)
+
+    @Deprecated(
+        message = "dfs is deprecated, use recursively instead.",
+        replaceWith = ReplaceWith("this.cols(predicate).recursively()"),
+        level = DeprecationLevel.WARNING,
+    )
+    public fun String.dfs(predicate: (ColumnWithPath<*>) -> Boolean): ColumnSet<*> =
+        columnGroup(this).dfs(predicate)
+
+    @Deprecated(
+        message = "dfs is deprecated, use recursively instead.",
```

Comment:
It gives me some idea: looks like we have no strict rules for removing deprecated stuff.
Could we somehow comment it with "it will be removed 0.13, for example).
Also, please find the file with String constants, containing deprecated messages and refactor all these String objects with the constants